src/app/core/pricing/pricing-tier.service.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from "@angular/core";
+import { BehaviorSubject, Observable } from "rxjs";
+
+@Injectable({
+    providedIn: "root"
+})
+export class PricingTierService {
+    private isVolumeDiscountActive$ = new BehaviorSubject(false);
+    toggleDiscount = () => this.isVolumeDiscountActive$.next(!this.isVolumeDiscountActive$.value);
```

Comment:
Using a getter resulted in `undefined` errors cropping up over the place due to the context of `this` ([more info here](https://stackoverflow.com/a/57588867)).

Reply:
Use ```Output``` to pass value from toggle upwards.

Reply:
I made the fix using @Output.

You can check out the code on the develop branch if you want to see how its done. I also removed the observable from the toggle-widget component. That way, the component remains "dumb" and only received true or false and emits a new value